src/commands/run.rs

```diff
@@ -0,0 +1,263 @@
+    let output_dir = output_dir.unwrap_or_else(|| {
+        let timestamp = chrono::Utc::now().format("%Y%m%d_%H-%M-%S");
+        PathBuf::from(format!("sprocket_runs/{}/{}", name, timestamp))
+    });
+
+    let output_dir = if output_dir.exists() {
+        if overwrite {
+            fs::remove_dir_all(&output_dir).with_context(|| {
+                format!(
+                    "failed to remove output directory `{dir}`",
+                    dir = output_dir.display()
+                )
+            })?;
+            output_dir
+        } else {
+            bail!(
+                "output directory `{dir}` already exists; use the `--overwrite` option to \
+                 overwrite it",
+                dir = output_dir.display()
+            );
+        }
+    } else {
+        output_dir
+    };
+
+    fs::create_dir_all(&output_dir).with_context(|| {
+        format!(
+            "failed to create output directory `{dir}`",
+            dir = output_dir.display()
+        )
+    })?;
+
+    log::info!(
+        "output directory created: `{dir}`",
+        dir = output_dir.display()
+    );
+
+    Ok(output_dir)
+}
+
+/// Runs a task.
+pub async fn run(args: RunArgs) -> Result<()> {
+    eprintln!(
+        "the `run` command is in alpha testing and does not currently support workflows or using \
+         containers."
+    );
+
+    let file = args.file;
+    if Path::new(&file).is_dir() {
+        anyhow::bail!("expected a WDL document, found a directory");
+    }
+    let (config, mut stream) = get_display_config(args.report_mode, args.no_color);
+
+    let results = analyze(&file, vec![], false, false).await?;
+
+    let uri = if let Ok(uri) = Url::parse(&file) {
+        uri
+    } else {
+        path_to_uri(&file).expect("file should be a local path")
+    };
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+    let document = result.document();
+
+    let mut engine = Engine::new(LocalTaskExecutionBackend::new());
+    let (path, name, inputs) = if let Some(path) = args.inputs {
+        let abs_path = absolute(&path).with_context(|| {
+            format!(
+                "failed to determine the absolute path of `{path}`",
+                path = path.display()
+            )
+        })?;
+        match Inputs::parse(document, &abs_path)? {
+            Some((name, inputs)) => (Some(path), name, inputs),
+            None => bail!(
+                "inputs file `{path}` is empty; use the `--name` option to specify the name of \
+                 the task or workflow to run",
+                path = path.display()
+            ),
+        }
+    } else if let Some(name) = args.name {
+        if document.task_by_name(&name).is_some() {
+            (None, name, Inputs::Task(Default::default()))
+        } else if document.workflow().is_some() {
```

Comment:
This doesn't actually check that `workflow` matches `name` does it? You could specify any `name` value and it would run whatever `workflow` was in the file.

Reply:
It does get checked below though. L248-250

Reply:
See other thread.